driver/src/main/java/org/neo4j/driver/internal/cursor/ResultCursorFactoryImpl.java

```diff
@@ -39,24 +39,25 @@
 
     private final PullResponseHandler pullHandler;
     private final PullAllResponseHandler pullAllHandler;
-    private final boolean waitForRunResponse;
     private final Message runMessage;
+    private final CompletableFuture<Void> runFuture;
```

Comment:
The purpose of that `runFuture` isn't clear to me at all just from looking at that bit.

Reply:
To avoid duplication, could you take a look at the following [message](https://github.com/neo4j/neo4j-java-driver/pull/897#issuecomment-852917461) please?

Reply:
Agreed the [approach](https://github.com/neo4j/neo4j-java-driver/pull/897#issuecomment-855839119).